Sprint_2

AnastasiyaIlyina opened the pull request:
Сделано основное и дополнительное задание.
Покрытие тестов 100%

AnastasiyaIlyina committed:
Сделано основное и дополнительное задание.
Покрытие тестов 100%

alesyaisachenko commented on src/test/java/com/example/AnimalTest.java:
```diff
@@ -0,0 +1,49 @@
+        this.expectedFoodListReturn = expectedFoodListReturn;
+    }
+
+    @Parameterized.Parameters
```
Параметризованные тесты лучше выносить в отдельный класс.
Сейчас у тебя не параметризованные тесты запускаются несколько раз подряд -- это трата ресурсов.

Для параметризованного теста создай, пожалуйста, отдельный тестовый класс.

Это нужно исправить.

AnastasiyaIlyina replied:
Исправлено

alesyaisachenko commented on src/test/java/com/example/AnimalTest.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    @Test
+    public void getFoodHerbivore_returnHerbivoreFoodList() throws Exception {
```
getFoodHerbivore_returnHerbivoreFoodList() -- не удачный нейминг тестового метода, нужно getFoodHerbivoreReturnHerbivoreFoodList().

в Java есть строгие правила к именованию пакетов, классов, методов, переменных. Шпаргалка по неймингу: - классы - имя начинается с заглавной буквы, если в имени несколько слов, каждое слово пишется с заглавной буквы слитно (пример: Praktikum, BestProgram) - методы - именуются lowerCamelCase - переменные - именуются lowerCamelCase, нельзя использовать цифры в качестве первого символа (пример: firstName, expectedResult, tylerTheCreator. ) - пакеты - допустимы только строчные буквы - константы - Константы состоят из заглавных символов, если в имени несколько слов то они разделяются нижним подчёркиванием "_".

Замечание относится ко всему коду.

Это нужно исправить.

AnastasiyaIlyina replied:
Исправлено

alesyaisachenko commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -0,0 +1,41 @@
+
+    @Test
+    public void eatMeat_ReturnPredatorsFoodList() throws Exception {
+        Feline feline = new Feline();
```
Feline feline = new Feline(); -- этот код можно было вынести в @Before.
Таким образом ты бы избежала дублирование кода.

Можно не править.

AnastasiyaIlyina replied:
Исправлено

AnastasiyaIlyina committed:
Правки по код ревью:
1. Параметризованный тест вынесен в отдельный класс
2. Наименования тестов исправлены
3. Добавлен @before там где необходимо
